src/classes/ADDR_AdminAcc_TEST.cls

```diff
@@ -0,0 +1,891 @@
+    static void deleteDefaultAddr() {
+        configSettings();
+            
+        // this creates a default Address for each Account
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(2, UTIL_Describe.getAdminAccRecTypeID());
+        List<Address__c> addrs = accsAddrs.addrs;
+       
+        // now let's delete the Addresses
+        system.assertEquals(false, ADDR_Addresses_TDTM.hasRunAddrTrigger);
+        Test.startTest();
+        delete addrs;
+        Test.stopTest();
+    
+        // verify that the Account address fields are cleared
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
+        BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude, is_Address_Override__c, Current_Address__c from Account]);
+        
+        for (Account acc : mapAccIdAcc.values()) {
+            system.assertEquals(null, acc.BillingStreet);
+            system.assertEquals(null, acc.BillingCity);
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Delete existing non-default override Addresses.  
+    verify: Account billing address fields not changed.
+    **********************************************************************************************************/            
+    @isTest
+    static void deleteNonDefaultOverrideAddr() {
+        configSettings();
+        
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(2, UTIL_Describe.getAdminAccRecTypeID());
+        List<Address__c> addrs = accsAddrs.addrs;
+        
+        // create additional addresses
+        addrs = UTIL_UnitTestData_TEST.initTestAddr(2);
+        
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            addrs[i].Parent_Account__c = accsAddrs.accs[i].Id;
+            addrs[i].Default_Address__c = false;
+            addrs[i].MailingStreet__c = 'override' + i;
+            addrs[i].MailingCity__c = 'override' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrs;
+        
+        // set the accounts' address overrides
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            Account acc = accsAddrs.accs[i];
+            acc.Current_Address__c = addrs[i].Id;
+            acc.is_Address_Override__c = true;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update accsAddrs.accs;        
+        
+        // now let's delete the override Addresses
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        delete addrs;
+        Test.stopTest();
+    
+        // verify that the Account address fields went back to the default
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.